### PR TITLE
[Self-service error codes] No redundant logging in Ledger API [DPP-731]

### PR DESCRIPTION
**Pull request by tudor-da**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/ValidationLogger.scala [hunk 1]
```diff
@@ -11,16 +12,10 @@ object ValidationLogger {
       loggingContext: LoggingContext,
   ): Throwable = {
     logger.debug(s"Request validation failed for $request. Message: ${t.getMessage}")
-    logger.info(t.getMessage)
-    t
-  }
-
-  def logFailureWithContext[Request, T <: Throwable](request: Request, t: T)(implicit
```
Comment (tudor-da) [inline ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/ValidationLogger.scala:18]: Duplicate with different name (compared to above)

File: ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/validation/ErrorFactories.scala [hunk 2]
```diff
@@ -52,18 +51,13 @@ class ErrorFactories private (errorCodesVersionSwitcher: ErrorCodesVersionSwitch
...
-      v1 = ValidationLogger.logFailureWithContext(
-        request,
-        io.grpc.Status.INVALID_ARGUMENT
-          .withDescription(message)
-          .asRuntimeException(),
-      ),
+      v1 = io.grpc.Status.INVALID_ARGUMENT
+        .withDescription(message)
+        .asRuntimeException(),
       v2 = LedgerApiErrors.ReadErrors.MalformedPackageId
```
Comment (tudor-da) [inline ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/validation/ErrorFactories.scala:61]: Simplification

File: ledger/error/src/main/scala/com/daml/error/definitions/PackageServiceError.scala [hunk 3]
```diff
@@ -19,7 +19,14 @@ import com.daml.lf.data.Ref.PackageId
 import com.daml.lf.engine.Error
 import com.daml.lf.{VersionRange, language, validation}
 
-trait PackageServiceError extends BaseError
+abstract class PackageServiceError(
+    override val cause: String,
+    override val throwableO: Option[Throwable] = None,
+)(implicit override val code: ErrorCode)
+    extends BaseError.Impl(cause, throwableO) {
+  final override def logOnCreation: Boolean = true
```
Comment (mziolekda) [inline ledger/error/src/main/scala/com/daml/error/definitions/PackageServiceError.scala:27]: That is a default on BaseError.Impl

Comment (tudor-da) [reply ledger/error/src/main/scala/com/daml/error/definitions/PackageServiceError.scala:27]: yes and I want to be sure it stays like that

Comment (tudor-da) [reply ledger/error/src/main/scala/com/daml/error/definitions/PackageServiceError.scala:27]: i.e. enforce that `PackageServiceError` logs on creation.

**Review by mziolekda — APPROVED** [review]

**Pull request merged**